docs/performance/inserts/methods.txt

```diff
@@ -0,0 +1,284 @@
+If your client supports query string parameter substitution, you can use the
+``UNNEST`` method with static prepared statements.
+
+For example, using the CrateDB Python client, the following is possible:
+
+.. code-block:: python
+
+    client.execute("""
+      INSERT INTO my_table (id, name)
+        (SELECT *
+           FROM UNNEST(?, ?))
+    """, ([1, 2, 3], ["Arthur", "Trillian", "Marvin"]))
+
+Here, you can vary the number of rows being inserted without having to change
+the prepared statement.
+
+.. _inserts_multiple_values:
+
+Multiple Value Expressions
+--------------------------
+
+You can insert multiple rows with multiple value expressions, like so:
+
+.. code-block:: psql
+
+    cr> INSERT INTO my_table (id, name)
+    ...      VALUES (1, 'Arthur'),
+    ...             (2, 'Trillian'),
+    ...             (2, 'Marvin');
+    INSERT OK, 3 rows affected  (... sec)
+
+This method of doing bulk inserts is usually slower than the ``UNNEST`` method,
+because parsing is more expensive. The query looks nicer for humans though.
+
+The only problem is that the structure of the insert statement is variable on
+the number of rows to insert. So, if you are inserting a variable number of
+rows, you have to prepare the SQL statement using some form of string
+concatenation each time.
+
+Query string parameter substitution is recommended over string concatenation,
+and so the ``UNNEST`` method is recommended over the multiple value expressions
+method.
+
+.. _inserts_client_approaches:
+
+Client Approaches
+=================
+
+There are three client approaches for doing inserts:
+
+- :ref:`Standard querying <inserts_standard_querying>`
+- :ref:`Bulk operations <inserts_bulk_operations>`
+- :ref:`Prepared statements <inserts_prepared_statements>`
+
+Each client approach can be used to insert :ref:`any type of insert statement
+<insert_statement_types>`.
+
+.. _inserts_standard_querying:
+
+Standard Querying
+-----------------
+
+The standard way of issuing insert statements executes one statement at a time
+and does not make use of :ref:`inserts_bulk_operations` or any special
+:ref:`inserts_prepared_statements` client feature.
+
+For example, using the CrateDB Python client, here's a :ref:`single insert
+<inserts_single_inserts>`:
+
+.. code-block:: python
+
+   client.execute("INSERT INTO my_table (column_a) VALUES (?)", ["value 1"])
+
+.. _inserts_bulk_operations:
+
+Bulk Operations
+---------------
+
+You can use the `bulk operations`_ feature of the CrateDB HTTP endpoint to
+perform many inserts in a single operation.
+
+The advantages are the same as using the :ref:`UNNEST method<inserts_unnest>`:
+
+- Significantly less internal network traffic than executing each insert
+  statement individually
+
+- Even though you're executing multiple insert statements, the bulk query only
+  needs to be parsed, planned, and executed once
+
+- If `translog.durability`_ is set to ``REQUEST`` (the default), a bulk insert
+  only flushes the disk once
```

Comment:
also here it should state `... once for every shard written to..`

Reply:
resolving with https://github.com/crate/guide/pull/108